src/http.ts

```diff
@@ -24,6 +24,7 @@ export interface ConnectionOptions {
   baseURL: string;
   sessionToken: string;
   source: string;
+  requestId?: string;
```

Comment:
`code-client` is a `client` :p I don't see any harm in it generating the `id`. That's what our `@snyk-correlation` does. If there is no `snyk-request-id` it generates it, doesn't matter if you are the first service, service in the middle, etc.

Reply:
`code-client` is kind of SDK for the actual client - the CLI. 

That is why we had at least 3 (for a big scan, it's much more) unique request IDs for the same request in deeproxy 😬 . It was meaningless.

But if now we create this from code-client and we keep the same UUID for all calls towards deeproxy, this make more sense 💪

Reply:
Ok, I am convinced now, for a while I was still thinking of `code-client` as being another service and creating request ID there was as "meaningful" as it was in `deeproxy`. But now its al clear, we are on the same page 🤝

Reply:
And I think it still makes sense to pass `requestId` from the `cli`, right? We just make sure here that whichever client uses `code-client`, we will have `requestId` out of the box.

Reply:
> it still makes sense to pass requestId from the cli

Yes of course, this is important to debug customer's issue. When they use CLI with `-d` flag, they get a requestId that we can track down through the stack. 

Only to create a requestId in code-client in case it's not provided.